### PR TITLE
Upgrade npm package dependencies

### PR DESCRIPTION
**Pull request by lalitpatel**
### Summary:
There are a bunch of dependencies that were EOL and losing support with node 14+. This PR upgrades all the npm package dependencies to the latest version. I am using node `v14.7.0`.

### Also:
* Fixed lint issues
* Mocked i18next for tests to fix warnings while executing tests.

#### Approach for mocking i18next:
This is the warning we get while running tests:
```
console.warn node_modules/react-i18next/dist/commonjs/utils.js:22
react-i18next:: You will need pass in an i18next instance by using initReactI18next
```
Current implimentation: https://github.com/i18next/react-i18next/issues/748
Alternate solutions: https://react.i18next.com/misc/testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,29 +4,29 @@
...
-    "i18next": "^15.0.7",
-    "i18next-browser-languagedetector": "^3.0.1",
+    "@types/express": "^4.17.7",
+    "@types/react-redux": "^7.1.9",
+    "@types/react-router-dom": "^5.1.5",
+    "@types/socket.io": "^2.1.10",
+    "@types/socket.io-client": "^1.4.33",
+    "@types/uuid": "^8.0.1",
+    "express": "^4.17.1",
+    "fsevents": "^2.1.3",
```
Comment (HSZemi) [inline package.json:14]: Where does this one come from? It's no bueno if you're not on MacOS:
```
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
```

Comment (lalitpatel) [reply package.json]: This should be fixed now. I had added `fsevents@2.1.3` to package.json to resolve an issue. Forgot to take it out.

You may have to remove your `node_modules` and `package-lock.json` and do a fresh `npm install` after setting node version to 14

```
rm -rf node_modules package-lock.json
npm install
```

File: src/index.tsx [hunk 2]
```diff
@@ -101,6 +103,8 @@ const createMySocketMiddleware = () => {
 
 const store: Store = createStore<ApplicationState, Action, any, Store>(updateState,
     {
+        iconStyle: initialIconStyleState,
+        replay: initialReplayState,
```
Comment (lalitpatel) [inline src/index.tsx:107]: This was done to address lint issues.

File: src/components/draft/Draft.tsx [hunk 3]
```diff
@@ -110,9 +110,9 @@ class Draft extends React.Component<IProps, IState> {
                 <div id="container" className="container is-fluid">
                     <div className="columns is-mobile">
                         <div className="column is-1 py-0">
-                                <a onClick={this.disconnectAndGoBack} aria-label="Go back" className="back-icon header-navigation">
+                                <button onClick={this.disconnectAndGoBack} aria-label="Go back" className="button is-text back-icon header-navigation">
                                     <KeyboardBackspaceIcon size={48} />
-                                </a>
+                                </button>
```
Comment (lalitpatel) [inline src/components/draft/Draft.tsx:115]: This was done to address lint warnings.

File: src/components/draft/PlayerOnlineStatus.tsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ class PlayerOnlineStatus extends React.Component<IProps, object> {
     }
 
     private isConnected() {
-        return this.isHost() && this.props.hostConnected || this.isGuest() && this.props.guestConnected;
+        return (this.isHost() && this.props.hostConnected) || (this.isGuest() && this.props.guestConnected);
```
Comment (lalitpatel) [inline src/components/draft/PlayerOnlineStatus.tsx:21]: This was also done to address lint warnings.

**Comment by HSZemi** [comment]
Thank you! :yellow_heart:

**Pull request merged**